### DISCUSSION TITLE
自定义编辑器在 tinyEngine里的editor要怎么配置?

### DISCUSSION THREAD

**Discussion by stwflyfox** [opentiny/tiny-engine#326] [Q&A]
配置为 Select 不起作用， 还有怎么配置 options 为 state 里的变量。
![1fc52883b219934e58ba944979eaf38](https://github.com/opentiny/tiny-engine/assets/3983204/8dc3fee1-2947-4890-86fe-731298e87090)
最终要实现的编辑器是这个下拉表格。
![477534e1031516e1843eed496920089](https://github.com/opentiny/tiny-engine/assets/3983204/c389122e-9327-4ddf-bdb5-5eae31e75c29)

**Comment by chilingling** [comment]
当前可以这样配置：
```
{
  "component": "select",
  "options": {
    "type": "JSExpression",
    "value": "this.state.testOptions"
  }
}
```
后续我们继续优化一下出码能力，方便用户编写

**Comment by stwflyfox** [comment]
![image](https://github.com/opentiny/tiny-engine/assets/3983204/bcaa2891-a957-4bc5-8a52-d0fd9c7622be)
那这个在editor里要怎么配置呢？

**Reply by chilingling** [reply]
如果要用 `ref` 跟  `v-model`，需要用编辑器插槽。

如果不用 `ref` 跟 `v-model`，则可以这样配置：
```
{
  "component": "select",
  "options": {
    "type": "JSExpression",
    "value": "this.state.testOptions"
  },
 "filterable": true,
 "filter-mothod":  {
    "type": "JSExpression",
    "value": "this.state.filter"
  },
  "clearable": true,
  “value-field”: "id",
  "text-field": "city",
  "render-type": "grid",
 "grid-op": {
  "type": "JSExpression",
    "value": "this.state.gridOpSingle"
  }
}
```

**Comment by stwflyfox** [comment]
editor用以下配置后， clearable 、 valueField、textField 没有效果， 感觉这个 select 是默认的编辑器， 并不是 TinySelect ， 所以没效果。
{
  "component": "select",
  "options": {
    "type": "JSExpression",
    "value": "this.state.unitList"
  },
  "clearable": true,
  "valueField": "UnitName",
  "textField": "UnitName"
}
而且配置方式也和TinyVue的示例方式不同：
<tiny-grid-column
        field="area"
        title="区域"
        :editor="{
          component: Select,
          attrs: { options, textField: 'name', valueField: 'id' }
        }"
        :format-config="{ data: options, label: 'name', value: 'id' }"
        format-text="enum"
      ></tiny-grid-column>

**Reply by chilingling** [reply]
要使用 TinySelect 的话，当前支持还有点问题，可以有两种支持方式：
1. 增加编辑器插槽（在 bundle.json 增加 编辑器插槽配置）
2. 出码增加对应解析

**Reply by stwflyfox** [reply]
1. 增加编辑器插槽碰到问题：
![image](https://github.com/opentiny/tiny-engine/assets/3983204/76df37ba-d482-4b23-9cd5-1d340dd3d29b)

2.出码增加对应解析的方式是不是改起来快些？ 自定义出码的alpha版本能否应用上？
